pkg/proto/configuration/grpc/grpc.proto

```diff
@@ -277,11 +277,67 @@ message TLSClientCertificateAuthenticationPolicy {
   // validate the remote TLS client.
   string client_certificate_authorities = 1;
 
-  // Authentication metadata to use.
+  // Was `metadata`, instead use `metadata_extraction_jmespath_expression`
+  // to provide static or dynamic metadata
+  reserved 2;
+
+  // This option allows specifying a boolean JMESPath expression that
+  // can be used to place additional requirements on alt names within
+  // the client certificates.
+  //
+  // The context data has the following structure:
+  //
+  // {
+  //    // Contains every DNS Subject Alt Name provided
+  //   "dnsNames": ["example.com", "*.sub.example.com"],
+  // 
+  //   // Contains every Email Subject Alt Name provided
+  //   "emailAddresses": ["me@example.com"] 
+  // }
+  //
+  // You could enforce that the DNS name matches an expection:
+  //
+  //     contains(dnsNames, "example.com")
+  //
+  // Or to enforce that an expected email addresses is presented
+  //
+  //     length(emailAddresses[?ends_with(@, '@example.com')]) > `0`
+  //
+  //
+  // In case no additional requirements need to be made, it is possible to
+  // use the following JMESPath expression (including the backticks):
+  //
+  //     `true`
+  //
+  // More details about JMESPath can be found at https://jmespath.org/.
+  //
+  // It is recommended that this feature is only used for making
+  // authentication decisions; not authorization decisions. The reason
+  // being that validation failures will cause gRPC requests to fail
+  // with code UNAUTHENTICATED; not PERMISSION_DENIED.
+  string validation_jmespath_expression = 3;
+
+  // JMESPath expression for converting Certificates information to a
+  // Protobuf message of type buildbarn.auth.AuthenticationMetadata.
+  // The resulting metadata can be used for purposes such as authorization.
+  //
+  // The context data has the following structure:
+  //
+  // {
```

Comment:
Instead of documenting the input data structure once again, I guess it's sufficient to just refer to the description above.